### PR TITLE
Add `SplitSegmentSorting`

### PR DESCRIPTION
**Pull request by khl02007**
per our discussion @alejoe91 @samuelgarcia: This is a new class that takes a sorting with a single segment and a list of recordings and returns a sorting with multiple segments, each defined by the length of the recordings in the list. This would be useful for getting back the times of the spikes after sorting a concatenated recording, which removes time information. Please take a look.

### CODE REVIEW

File: spikeinterface/core/segmentutils.py [hunk 1]
```diff
@@ -264,5 +267,43 @@ def get_unit_spike_train(self, *args, **kwargs):
...
+    Parameters
+    ----------
+    parent_sorting : BaseSorting
+        Sorting with a single segment (e.g. from sorting concatenated recording)
+    recording_list : list
+        List of recordings whose lengths will be used to split the sorting
+        into smaller segments
+    """
+    def __init__(self, parent_sorting: BaseSorting, recording_list: List[BaseRecording]):
+        sampling_frequency = parent_sorting.get_sampling_frequency()
```
Comment (alejoe91) [inline spikeinterface/core/segmentutils.py:286]: Can you add an assertion that the input sorting is mono segment?

File: spikeinterface/core/segmentutils.py [hunk 2]
```diff
@@ -264,5 +267,43 @@ def get_unit_spike_train(self, *args, **kwargs):
 def append_sortings(*args, **kwargs):
     return AppendSegmentSorting(*args, **kwargs)
 
-
 append_sortings.__doc__ == AppendSegmentSorting.__doc__
+
+# Q: should this take as input a concatenated recording or just a list of recordings?
```
Comment (alejoe91) [inline spikeinterface/core/segmentutils.py:272]: I think we should have 3 options:
- concatenated recording
- recording_list
- None: in this case we can check if the sorting has a registered recording and if that is a concatenated one

File: spikeinterface/core/segmentutils.py [hunk 3]
```diff
@@ -264,5 +267,43 @@ def get_unit_spike_train(self, *args, **kwargs):
...
+class SplitSegmentSorting(BaseSorting):
+    """Splits a sorting with a single segment to multiple segments
+    based on the given list of recordings (must be in order)
+
+    Parameters
+    ----------
+    parent_sorting : BaseSorting
+        Sorting with a single segment (e.g. from sorting concatenated recording)
+    recording_list : list
```
Comment (alejoe91) [inline spikeinterface/core/segmentutils.py:281]: You can call this `recording_or_recording_list`

**Review by alejoe91 — CHANGES_REQUESTED** [review]

**Comment by samuelgarcia** [comment]
Looks good! 
Thanks a lot.

File: spikeinterface/core/segmentutils.py (hunk 2 above)
Comment (khl02007) [reply spikeinterface/core/segmentutils.py:272]: Done but please see my changes and comment

File: spikeinterface/core/segmentutils.py (hunk 3 above)
Comment (khl02007) [reply spikeinterface/core/segmentutils.py:281]: Done

File: spikeinterface/core/segmentutils.py (hunk 1 above)
Comment (khl02007) [reply spikeinterface/core/segmentutils.py:286]: Done

**Comment by khl02007** [comment]
By the way, I think the users should always give to sorters an AppendRecording and get back a SplitSorting. For the sorters that only operate on ConcatRecording (and not on AppendRecording) they should just convert the AppendRecording to ConcatRecording under the hood. This is because AppendRecording/SplitSorting is obviously the correct way to handle multiple segments. Maybe can print a message to let the user know that this is happening. Thoughts? @alejoe91 @samuelgarcia

File: spikeinterface/core/segmentutils.py [hunk 4]
```diff
@@ -264,5 +267,56 @@ def get_unit_spike_train(self, *args, **kwargs):
...
+
+class SplitSegmentSorting(BaseSorting):
+    """Splits a sorting with a single segment to multiple segments
+    based on the given list of recordings (must be in order)
+
+    Parameters
+    ----------
+    parent_sorting : BaseSorting
+        Sorting with a single segment (e.g. from sorting concatenated recording)
+    recording_list : list of recordings, ConcatenateSegmentRecording, or None
```
Comment (alejoe91) [inline spikeinterface/core/segmentutils.py:280]:
```suggestion
    recording_or_recording_list : list of recordings, ConcatenateSegmentRecording, or None
```

File: spikeinterface/core/segmentutils.py [hunk 5]
```diff
@@ -264,5 +267,56 @@ def get_unit_spike_train(self, *args, **kwargs):
...
+        If list of recordings, uses the lengths of those recordings to split the sorting
+        into smaller segments
+        If ConcatenateSegmentRecording, uses the associated list of recordings to split
+        the sorting into smaller segments
+        If None, looks for the recording associated with the sorting
+    """
+    def __init__(self, parent_sorting: BaseSorting, recording_or_recording_list):
```
Comment (alejoe91) [inline spikeinterface/core/segmentutils.py:287]:
```suggestion
    def __init__(self, parent_sorting: BaseSorting, recording_or_recording_list=None):
```

**Comment by alejoe91** [comment]
> By the way, I think the users should always give to sorters an AppendRecording and get back a SplitSorting. For the sorters that only operate on ConcatRecording (and not on AppendRecording) they should just convert the AppendRecording to ConcatRecording under the hood. This is because AppendRecording/SplitSorting is obviously the correct way to handle multiple segments. Maybe can print a message to let the user know that this is happening. Thoughts? @alejoe91 @samuelgarcia

I think it's a fair point. Can you open an issue so we can discuss there?

I made some minor changes and set default recording value to None. Good to go for me @samuelgarcia

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**